flink-python/pyflink/common/serialization.py

```diff
@@ -365,3 +365,27 @@ def __init__(self, record_class: str = None, avro_schema_string: str = None):
             j_serialization_schema = JAvroRowSerializationSchema(avro_schema_string)
 
         super(AvroRowSerializationSchema, self).__init__(j_serialization_schema)
+
+
+class Encoder(object):
+    """
+    A `Encoder` is used by the streaming file sink to perform the actual writing
+    of the incoming elements to the files in a bucket.
+    """
+
+    def __init__(self, j_encoder):
+        self.j_encoder = j_encoder
```

Comment:
For extensibility, it would be better to support passing a fully-qualified class name of user defined java encoder. In the future, we can provide such a common util for this series of constructors that uses either can pass a string or a java object.